src/leader.py

```diff
@@ -152,6 +158,77 @@ def lead(self):
             self.turn(code, position)
 
 
+    def drive_in_circle(self):
+        """Drives in a circle."""
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.turn_left()
+
+
+    def drive_wide_winding(self):
+        """Drives in a wide, snake-like pattern."""
+        self.demo_stop_flag = False
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.turn_right()
+        self.demo_turn_wait()
+
+        if self.demo_stop_flag: return
+
+        self.turn_left()
+        self.demo_turn_wait()
+
+    def drive_winding(self):
+        """Drives in a snake-like pattern."""
+        self.demo_stop_flag = False
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.fw.turn(110)
+        self.demo_turn_wait()
+
+        if self.demo_stop_flag: return
+
+        self.fw.turn(70)
+        self.demo_turn_wait()
+
+
+    def stop_demo_check(self):
+        """
+        Checks if the demo should stop.
+
+        Changes the demo_stop_flag based on any controller input recieved for
+        .1 consecutive seconds. If any button has been pressed for over .1
+        consecutive seconds demo_stop_flag will be set to true. If any button is
+        not being pressed the stop_count is reset.
+        """
+        code, position = self._controller.get_input()
+        sleep(.01)
+        if position is 1:
+            self.demo_stop_count += 1
+            if self.demo_stop_count > 10:
```

Comment:
I also wouldn't check for a button to be pressed 10+ times. I'm pretty sure if a button is pressed it only gets recorded once, not once per cycle but testing would need to be done to make sure.

Reply:
Not pressed 10+ times. demo_stop_count records the amount of times every time code, position = self._controller.get_input() is called when the position value is 1, and every time it calls that method again it grabs a new controller state. the button only needs to be pressed for .1 second to set the flag to break the while loop.

Reply:
So from what I understand, you're checking to see if they're holding the button for one second?